src/main/resources/include/hail/CommonDefs.h

```diff
@@ -0,0 +1,26 @@
+#ifndef HAIL_COMMONDEFS_H
+#define HAIL_COMMONDEFS_H 1
+//
+// A few useful macros
+//
+
+// A way to begin/end namespaces without making a text editor change your
+// indentation level.
+
+#define NAMESPACE_BEGIN(_name) namespace _name {
+#define NAMESPACE_END(_name)   }
```

Comment:
I would prefer to write the namespace in place.  I'm not aware of any editor we use that can't be configured to not indent within namespaces.  I use the following in emacs:

```
(defconst my-cc-style
  '("cc-mode"
    (c-offsets-alist . ((innamespace . -)))))
(c-add-style "my-cc-style" my-cc-style)
(add-hook 'c++-mode-hook (lambda () (c-set-style "my-cc-style")))
```